ext/intl/tests/gh12243.phpt

```diff
@@ -0,0 +1,24 @@
+--TEST--
+GitHub #12043 segfault with IntlDateFormatter::dateType where it equals to UDAT_PATTERN (icu 50) but
+IntldateFormatter::timeType needs to be set as such.
+--EXTENSIONS--
+intl
+--FILE--
+<?php
+
+$datetime = new \DateTime('2017-05-12 23:11:00 GMT+2');
+static $UDAT_PATTERN = -2;
```

Comment:
Feels like this constant should also be defined in the future?

Reply:
agreed